### PR TITLE
fix grab.js when using the spring action

### PR DESCRIPTION
**Pull request by AndrewMeadows**
grab.js was not behaving well after being converted to use the **spring** action.

The problem was that the action's instance was being thrashed (constantly created and deleted) by old updates coming from the entity-server.  The action would be created locally and then sent up to the entity-server, however if there were an incoming update from the entity-server then the action would be deleted when it was not found inside the action data -- its absence is interpreted as a deletion.  Meanwhile the update echo from the entity-server from the creation of the action would eventually re-create it, or maybe the grab.js script would re-create it when trying to update it by **id**.

Similarly there would be thrashing when the grabbed object was let go while moving -- the script would delete the action but unprocessed entity-server updates would recreate it, then still later updates would delete it, etc.  Sometimes the final result was that the action was deleted, but sometimes it would survive the cycles creating a "leaked" action that the script had forgotten about, and the object would now be pinned in place and future grabs would fight new action against old (who would win depended on how they were ordered in Bullet's list of actions -- typically the newer action would win).

The solution is to ignore entity-server updates on actions when the local interface owns the simulation.  This introduces a problem where multiple participants cannot edit actions simultaneously -- the owner of the simulation will ignore updates from others and re-assert their view of the object's state to the entity-server.  We'll have to live with this wart until we overhaul how the action properties are created/updated/deleted (e.g. maybe they should be REST-ful, and maybe we need server-side permission filters for some operations on actions).

### CODE REVIEW

File: examples/grab.js [hunk 1]
```diff
@@ -43,9 +43,10 @@ var gMaxGrabDistance;
...
 var gGrabMode = "xzplane";
 
-// gGrabOffset allows the user to grab an object off-center.  It points from ray's intersection
-// with the move-plane to object center (at the moment the grab is initiated).  Future target positions
-// are relative to the ray's intersection by the same offset.
+// gGrabOffset allows the user to grab an object off-center.  It points from the object's center
+// to the point where the ray intersects the grab plane (at the moment the grab is initiated).
+// Future target positions of the ray intersection are on the same plane, and the offset is subtracted
+// to compute the target position of the object's center.
 var gGrabOffset = { x: 0, y: 0, z: 0 };
```
Comment (AndrewMeadows) [inline examples/grab.js]: I'm changing the direction of the **gGrabOffset** vector to make more sense.  It now points from object center to the grab point.

Comment (ZappoMan) [inline examples/grab.js]: why does this variable name have a "g" on it? that's contrary to our coding standard.

File: examples/grab.js [hunk 2]
```diff
@@ -287,6 +288,7 @@ function mouseMoveEvent(event) {
             var pointOnCylinder = Vec3.multiply(planeNormal, gXzDistanceToGrab);
             pointOnCylinder = Vec3.sum(Camera.getPosition(), pointOnCylinder);
             newTargetPosition = mouseIntersectionWithPlane(pointOnCylinder, planeNormal, event);
+            gPointOnPlane = Vec3.sum(newTargetPosition, gGrabOffset);
```
Comment (AndrewMeadows) [inline examples/grab.js]: Updating **gPointOnPlane** here fixes a bug when transitioning from vertical to horizontal grab mode -- the new horizontal plane will be updated based on the object's current position.

File: libraries/entities/src/EntityItem.cpp [hunk 3]
```diff
@@ -702,7 +704,16 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
+    {   // When we own the simulation we don't accept updates to the entity's actions
+        // but since we're using macros below we have to temporarily modify overwriteLocalData.
+        // NOTE: this prevents userB from adding an action to an object1 when UserA 
+        // has simulation ownership of it.
+        // TODO: figure out how to allow multiple users to update actions simultaneously
+        bool oldOverwrite = overwriteLocalData;
+        overwriteLocalData = overwriteLocalData && !weOwnSimulation;
+        READ_ENTITY_PROPERTY(PROP_ACTION_DATA, QByteArray, setActionData);
+        overwriteLocalData = oldOverwrite;
+    }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: Ignoring entity-server action updates for "owned" objects fixes the main problem when grabbing with actions.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 69db9aa01ac18f1330396f9b4fd8d7f8d847fc78 available at: https://gist.github.com/bf8b62a8ad5bf924c927<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5320.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5320.dmg

File: examples/grab.js (hunk 1 above)
Comment (AndrewMeadows) [inline examples/grab.js]: **g** indicates that it is a global variable rather than local.  I could fix it but would like to do it in a separate PR since it would change 50% of the file and obfuscate the pertinent changes here.

Comment (ZappoMan) [inline examples/grab.js]: ok - but we should follow the coding standard for JS too... which means no "hungarian notation"...
